src/System.Linq/src/System/Linq/Select.SpeedOpt.cs

```diff
@@ -171,7 +171,7 @@ private sealed partial class SelectRangeIterator<TResult> : Iterator<TResult>, I
             public SelectRangeIterator(int start, int end, Func<int, TResult> selector)
             {
                 Debug.Assert(start < end);
-                Debug.Assert((end - start) <= int.MaxValue);
+                Debug.Assert(unchecked((uint)(end - start) <= (uint)int.MaxValue));
```

Comment:
The `unchecked` isn't needed. `unchecked` is the default.

Reply:
Just make easy to read. Many other places used `unchecked` too.

Reply:
Done.